Microsoft.Toolkit.Uwp.UI/Extensions/TextBox/TextBoxExtensions.Regex.Internals.cs

```diff
@@ -70,6 +70,12 @@ private static void ValidateTextBox(TextBox textBox, bool force = true)
                 case ValidationType.Decimal:
                     regexMatch = textBox.Text.IsDecimal();
                     break;
+                case ValidationType.Currency:
+                    // @"(^\d*\.\d{2}$)" regex pattern to detect currency sign with currency value
+                    // Mathes: $100.00, $100, $10.25
+                    // Non-Matches: 100., $10.233, $10.
+                    regexMatch = Regex.IsMatch(textBox.Text, @"(^\d*\.\d{2}$)");
```

Comment:
We should handle currency symbols and decimal separators other than USD. Luckily, dotnet has this functionality built in and we don't need to fiddle with Regex.

[System.Globalization.NumberFormatInfo](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.numberformatinfo) should have everything we need for this.

Firing up an empty console app, here's what we can do.
```csharp

string FormatCurrency(string value)
{
    // Parse the entered data into a decimal, allowing existing currency symbols.
    var amount = decimal.Parse(value, NumberStyles.Currency);

    // Format the decimal into a human readable string for the current culture.
    return amount.ToString("C2", CultureInfo.CurrentCulture);
}

// Stress testing USD
FormatCurrency("500"); // $500.00
FormatCurrency("5"); // $5.00
FormatCurrency("5.00"); // $5.00
FormatCurrency(".05"); // $0.05
FormatCurrency(".5"); // $0.50
FormatCurrency("$.5"); // $0.50

// With other cultures
FormatCurrency("5"); // nl-NL, € 5,00
FormatCurrency("5"); // fr-FR 5,00 €
FormatCurrency("5"); // ja-JP ¥5.00
FormatCurrency("5"); // en-GB £5.00
```